File: MobileBackend/tutoring-pojo/src/main/java/com/tutoring/entity/Lesson.java
package com.tutoring.entity;

import com.baomidou.mybatisplus.annotation.*;
import lombok.Builder;
import lombok.*;

import java.io.Serializable;
import java.time.LocalDateTime;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Builder
@ToString
@TableName("lessons")
public class Lesson implements Serializable {

    @TableId(value = "lesson_id", type = IdType.AUTO)
    private Long lessonId;

    @TableField("course_id")
    private Long courseId;

    @TableField("title")
    private String title;

    @TableField("content")
    private String content;

    @TableField("image_urls")
    private String imageUrls;

    @TableField("pdf_urls")
    private String pdfUrls;

    @TableField(value = "created_at", fill = FieldFill.INSERT)
    private LocalDateTime createdAt;

    @TableField(value = "updated_at", fill = FieldFill.INSERT_UPDATE)
    private LocalDateTime updatedAt;

    @TableField(value = "is_completed", fill = FieldFill.INSERT_UPDATE)
    private boolean isCompleted;
}

File: TutoringFrontend/app/src/main/java/com/example/tutoring/data/Lesson.kt
package com.example.tutoring.data

data class Lesson(
    val completed: Boolean = false,
    val content: String = "",
    val courseId: Int = 0,
    val createdAt: String? = "",
    val imageUrls: String = "",
    val lessonId: Int? = 0,
    val pdfUrls: String = "",
    val title: String = "",
    val updatedAt: String? = ""
)
